internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -185,6 +222,190 @@ func (a *runTaskAction) Do(o *Orchestrator) error {
 	return nil
 }
 
+// setupProxyConnections creates proxy connections to a.hosts in pauseContainer.
+// It assumes that pauseContainer is already running. A unique proxy connection
+// is created for each host (in parallel) using AWS SSM Port Forwarding through
+// a.remoteContainerID. Then, each connection is assigned an IP from a.network,
+// starting at the bottom of the IP range. Using iptables, TCP packets destined
+// for the connection's assigned IP are redirected to the connection. Finally,
+// the host's name is mapped to its assigned IP in /etc/hosts.
+func (o *Orchestrator) setupProxyConnections(ctx context.Context, pauseContainer string, a *runTaskAction) error {
+	fmt.Printf("\nSetting up proxy connections...\n")
+	g, gctx := errgroup.WithContext(ctx)
+	ports := make(map[Host]string)
+	portsMu := &sync.Mutex{}
+
+	for _, host := range a.hosts {
+		host := host
+		o.wg.Add(1)
+		g.Go(func() error {
+			defer o.wg.Done()
+			port, err := o.setupProxyConnection(gctx, pauseContainer, a.remoteContainerID, host)
+			if err != nil {
+				return fmt.Errorf("setup proxy connection for %q: %w", host.Name, err)
+			}
+
+			portsMu.Lock()
+			defer portsMu.Unlock()
+			ports[host] = port
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return err
+	}
+
+	ip := a.network.IP
+	increment := func(ip net.IP) (net.IP, error) {
+		// make a copy of the previous ip
+		tmp := ip.To4()
+		ip = net.IPv4(tmp[0], tmp[1], tmp[2], tmp[3])
+		ip = ip.To4()
+
+		var inc func(idx int) error
+		inc = func(idx int) error {
```

Comment:
```suggestion
		inc := func(idx int) error {
```

Reply:
sadly can't call it recursively if we do this 😔

Reply:
Is this another weird language quirk like `host := host`? 👀